src/client.ts

```diff
@@ -1572,17 +1572,6 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
       channels.push(c);
     }
 
-    if (!offlineMode) {
-      // If the channels are coming from server, then clear out the
-      // previously help offline channels.
-      for (const key in this.activeChannels) {
-        const channel = this.activeChannels[key];
-        if (channel.offlineMode) {
-          delete this.activeChannels[key];
-        }
-      }
-    }
-
```

Comment:
It seems like the removal of this part is unnecessary.. because `channel.offlineMode` should be false when a successful BE query happens.